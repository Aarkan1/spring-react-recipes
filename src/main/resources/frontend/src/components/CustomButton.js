import React, { useState, useEffect } from 'react'

export default function CustomButton(props) {
  const [counter, setCounter] = useState(0)

  useEffect(() => {
    setCounter(5)

    return () => {
      console.log('Unmounting funky component');
    }
  }, [])

  const incrementCount = () => {
    setCounter(counter + 1)
  }

  return (
    <div style={{
        display: 'inline-block',
        border: '1px solid gray',
        padding: '10px',
        margin: '10px',
        userSelect: 'none',
        cursor: 'pointer'
      }}
      onClick={incrementCount}
     >
       {props.label} count: {counter}
    </div>
  )
}




