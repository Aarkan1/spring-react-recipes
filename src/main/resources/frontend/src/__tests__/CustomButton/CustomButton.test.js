import React from 'react'
import ReactDOM from 'react-dom'
import CustomButton from '../../components/CustomButton'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('mounting without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<CustomButton />, div)
})

it('renders button correctly', () => {
  const { getByTestId } = render(<CustomButton label="click me please" />)
  expect(getByTestId('custom-button')).toHaveTextContent('click me please')
})

test('renders button correctly', () => {
  const { getByTestId } = render(<CustomButton label="save" />)
  expect(getByTestId('custom-button')).toHaveTextContent('save')
})

test('matches snapshot', () => {
  const tree = renderer.create(<CustomButton label="click me please" />).toJSON()
  expect(tree).toMatchSnapshot()
})