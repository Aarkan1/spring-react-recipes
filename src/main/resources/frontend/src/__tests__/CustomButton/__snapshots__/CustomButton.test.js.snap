// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="custom-button"
  style={
    Object {
      "border": "1px solid gray",
      "margin": "10px",
      "padding": "10px",
    }
  }
>
  click me please
</div>
`;
